Login and Registration UI

yugantarjain opened the pull request:
### Description
The UI for login and registration has been developed.

Fixes #7 

### Type of Change:
- Code
- User Interface
- Documentation

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

**Mock-ups**
Login | Sign Up
----- | -----
<img width="491" alt="Screen Shot 2020-06-03 at 1 45 28 PM" src="https://user-images.githubusercontent.com/31164725/83612981-c5329b80-a5a0-11ea-8e45-8818fbb5362b.png"> | <img width="491" alt="Screen Shot 2020-06-03 at 1 46 08 PM" src="https://user-images.githubusercontent.com/31164725/83613010-ccf24000-a5a0-11ea-86ac-d9196b11fce8.png">
<img width="491" alt="Screen Shot 2020-06-03 at 1 46 04 PM" src="https://user-images.githubusercontent.com/31164725/83613032-d8de0200-a5a0-11ea-98a4-521e8a5e7645.png"> | <img width="491" alt="Screen Shot 2020-06-03 at 1 49 04 PM" src="https://user-images.githubusercontent.com/31164725/83613152-032fbf80-a5a1-11ea-8753-f964ea0797e0.png">



### How Has This Been Tested?
Has been tested on Simulator.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] Any dependent changes have been published in downstream modules

yugantarjain committed:
Login UI done.

yugantarjain committed:
login and registration UI done

annabauza reviewed (changes requested):
every single ui PR needs a screenshots

vatsalkul commented:
Hey @yugantarjain I have following concerns about the implementation of UI.
1. When dismissing the keyboard by tapping return button the textfields become empty. (Implement a keyboard handler)
2. Size of textfields are too small (Tested on iPhone XR)
3. There is a default text in password fields.
4. According to Apple’s HIG “Always include a button that dismisses the modal view” i.e.  Add a top navigation bar button which says “Cancel” in SignUp View and add “signup” text in that nav bar.
5. Try to use systemPurple colours throughout the app.

yugantarjain commented:
1. Yes I know. I kept this PR strictly for UI.
2. I agree. Also, I have an alternate design idea too, please check it out in our design doc.
3. Yeah, again, kept it strictly for UI and didn't used property wrappers in this PR.
4. Will do. Also, I was thinking that a full screen view for Sign Up would be better than a sheet, what do you think?
5. Okay

sunjunkie commented:
My notes: 
- I think a full screen is better for sign up than a sheet.
- Also, the user must accept the terms explicitly. This means that the initial setting of the control must be off.
- I think for the UI PR, controls should appear to work even though the underlying action is not implemented. (e.g. the segmented control should change focus when tapped.)
- In the new screen design in the doc, what does "change" mean?
- Color decisions should be isolated.

annabauza commented:
Is the status bar has different color on both screens?

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+     @State private var showSignUpPage = false
+     
+     var body: some View {
+          VStack(spacing: 60) {
```
export this gutter to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+               Image("mentorship_system_logo")
+                    .resizable()
+                    .scaledToFit()
+                    .padding(.top, 40)
```
export this padding to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+                    .scaledToFit()
+                    .padding(.top, 40)
+               
+               VStack(spacing: 20) {
```
export this gutter to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+     
+     var body: some View {
+          VStack(spacing: 60) {
+               Image("mentorship_system_logo")
```
define the const as global access asset string

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+               VStack(spacing: 30) {
+                    Button(action: /*@START_MENU_TOKEN@*/{}/*@END_MENU_TOKEN@*/) {
+                         Text("Login")
+                              .padding([.vertical], 8)
```
export this padding to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+                              .padding([.vertical], 8)
+                              .padding([.horizontal], 40)
+                              .background(Color.blue)
+                              .foregroundColor(.white)
```
export this color to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+                         Text("Login")
+                              .padding([.vertical], 8)
+                              .padding([.horizontal], 40)
+                              .background(Color.blue)
```
export this color to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+                    Button(action: /*@START_MENU_TOKEN@*/{}/*@END_MENU_TOKEN@*/) {
+                         Text("Login")
+                              .padding([.vertical], 8)
+                              .padding([.horizontal], 40)
```
export this padding to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,57 @@
+                    }
+                    
+                    Button.init(action: { self.showSignUpPage.toggle() }) {
+                         Text("Signup")
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+struct SignUpView: View {
+     fileprivate func inputFieldsView() -> some View {
+          return VStack {
+               TextField("Name", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+     fileprivate func inputFieldsView() -> some View {
+          return VStack {
+               TextField("Name", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
+               TextField("Username", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+          return VStack {
+               TextField("Name", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
+               TextField("Username", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
+               TextField("Email", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+               TextField("Name", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
+               TextField("Username", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
+               TextField("Email", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
+               SecureField("Password", text: /*@START_MENU_TOKEN@*/ /*@PLACEHOLDER=Value@*/.constant("Apple")/*@END_MENU_TOKEN@*/)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+               TextField("Username", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
+               TextField("Email", text: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant("")/*@END_MENU_TOKEN@*/)
+               SecureField("Password", text: /*@START_MENU_TOKEN@*/ /*@PLACEHOLDER=Value@*/.constant("Apple")/*@END_MENU_TOKEN@*/)
+               SecureField("Confirm Password", text: /*@START_MENU_TOKEN@*/ /*@PLACEHOLDER=Value@*/.constant("Apple")/*@END_MENU_TOKEN@*/)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+     
+     fileprivate func availabilitySelectionView() -> some View {
+          return VStack {
+               Text("Available to be a:").font(.headline)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+               Text("Available to be a:").font(.headline)
+               
+               Picker(selection: .constant(2), label: Text("")) {
+                    Text("Mentor").tag(1)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+               
+               Picker(selection: .constant(2), label: Text("")) {
+                    Text("Mentor").tag(1)
+                    Text("Mentee").tag(2)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+               Picker(selection: .constant(2), label: Text("")) {
+                    Text("Mentor").tag(1)
+                    Text("Mentee").tag(2)
+                    Text("Both").tag(3)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+               VStack(spacing: 20) {
+                    
+                    //Sign Up heading
+                    Text("Sign Up").font(.largeTitle)
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+                    
+                    //sign up button
+                    Button(action: /*@START_MENU_TOKEN@*/{}/*@END_MENU_TOKEN@*/) {
+                         Text("Sign Up")
```
export this as global const kept in one file

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+                    //sign up button
+                    Button(action: /*@START_MENU_TOKEN@*/{}/*@END_MENU_TOKEN@*/) {
+                         Text("Sign Up")
+                              .padding([.vertical], 8)
```
export this padding to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+                    Button(action: /*@START_MENU_TOKEN@*/{}/*@END_MENU_TOKEN@*/) {
+                         Text("Sign Up")
+                              .padding([.vertical], 8)
+                              .padding([.horizontal], 40)
```
export this padding to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+                         Text("Sign Up")
+                              .padding([.vertical], 8)
+                              .padding([.horizontal], 40)
+                              .background(Color.blue)
```
export this color to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+                              .padding([.vertical], 8)
+                              .padding([.horizontal], 40)
+                              .background(Color.blue)
+                              .foregroundColor(.white)
```
export this color to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+                              .padding([.horizontal], 40)
+                              .background(Color.blue)
+                              .foregroundColor(.white)
+                              .cornerRadius(8)
```
export this radius to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+                    tncConsentView()
+               }
+          }
+          .background(Color(UIColor.systemGroupedBackground))
```
export this color to style/theme struct for whole app

annabauza commented on mentorship ios/Registration/SignUpView.swift:
```diff
@@ -0,0 +1,91 @@
+     fileprivate func tncConsentView() -> some View {
+          return HStack {
+               Toggle(isOn: /*@START_MENU_TOKEN@*//*@PLACEHOLDER=Value@*/.constant(true)/*@END_MENU_TOKEN@*/) {
+                    Text("I confirm that I have read and accept to be bound by the AnitaB.org Code of Conduct, Terms, and Privacy Policy. Further, I consent to the use of my information for the stated purpose.")
```
export this text to global const

yugantarjain committed:
code formatting improved. Progressed in refactoring and reusable components.

yugantarjain committed:
login UI improved.

yugantarjain committed:
registration UI enhanced. To fix: button placing

yugantarjain committed:
Login and sign up screen UI updated.

yugantarjain commented:
I've updated the PR. Update includes:
1. Refactoring and exporting to global constants.
2. Reusable views for custom TextField and SecureField.
3. Design changes, updated look and feel.
4. Have used property wrappers; UI elements are functional now.
5. Input validation done to enable and disable buttons.
6. Explicit dismiss button added in sign up page.

sunjunkie commented:
@yugantarjain Let's use the SwiftUI StyleConfiguration protocols to set the ButtonStyleConfiguration. We can also move things like "preferredCornerRadius" into the button style definition and just override it when setting the button configuration. Something like:

`struct AppButtonStyle: ButtonStyle {
    //any element we want to allow custom styling for a specific case goes here
    var backgroundColor: UIColor = .systemPurple
    var radius:CGFloat = 8
    var disabled: Bool = false

    public func makeBody(configuration: AppButtonStyle.Configuration) -> some View {

        configuration.label
              .padding([.vertical], 8)
              .padding([.horizontal], 40)
              .background(Color(backgroundColor))
              .foregroundColor(.white)
              .cornerRadius(radius)

    }
}
`
then 

`Button(action: {} {
                Text("Button")
            }
            .buttonStyle(AppButtonStyle(backgroundColor:.systemBlue))`

yugantarjain committed:
Custom reusable button implemented as button style.

yugantarjain commented:
@sunjunkie done.

yugantarjain committed:
TextFieldStyle developed. Global constants enhanced and encapsulated in structs. Form validation updated.

yugantarjain committed:
Localization added.

yugantarjain commented:
Updated. 

1. TextFieldStyle developed.
2. Global constants organized and implemented using structs.
3. Localization enabled.
4. Form validation improved.

annabauza commented on mentorship ios/Registration/LoginView.swift:
```diff
@@ -0,0 +1,67 @@
+//  LoginView.swift
+//  mentorship ios
+//
+//  Created by Yugantar Jain on 01/06/20.
```
@isabelcosta @mayburgos what should be here?

isabelcosta replied:
That line does not need to be there in my opinion. We even taken that out on Android app I think. these files are live, changed but the community so for me it does not make sense to have the file creator, since this appears on git @annabauza

sunjunkie replied:
There is also a copyright line inserted by xcode @annabauza @isabelcosta 
" //  Copyright © 2020 Yugantar Jain. All rights reserved."

isabelcosta replied:
then it could also be removed, right?

annabauza replied:
yes!

yugantarjain committed:
Design Constants: Padding restructured into screen and form.
New modifier to automatically add all the required paddings added.

yugantarjain committed:
Big strings shifted to constants file.

yugantarjain committed:
Final Polishing. Constants updated, sign up terms and conditions view improved, refactoring done, padding modifier updated.

isabelcosta commented:
This is awesome @yugantarjain 👏 🎉 
Sooo nice to see this UI!

yugantarjain commented:
@isabelcosta Thank you!! I'm very glad you like it!
